File: apps/commerce-api/src/main/java/com/loopers/domain/inventory/InventoryCommand.java
package com.loopers.domain.inventory;

import java.util.List;

public class InventoryCommand {
    public record Deduct(List<Option> options) {
        public List<Long> productIds() {
            return options.stream().map(Option::productId).toList();
        }

        public List<Long> productOptionIds() {
            return options.stream().map(Option::productOptionId).toList();
        }
    }

    public record Reverse(List<Option> options) {
        public List<Long> productIds() {
            return options.stream().map(Option::productId).toList();
        }

        public List<Long> productOptionIds() {
            return options.stream().map(Option::productOptionId).toList();
        }
    }

    public record Option(Long productId, Long productOptionId, Integer quantity) {

    }
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/domain/inventory/InventoryCommand.kt
package com.loopers.domain.inventory

class InventoryCommand {
    data class Deduct(
        val options: List<Option>
    ) {
        fun productIds(): List<Long> {
            return options.map { it.productId }.toList()
        }

        fun productOptionIds(): List<Long> {
            return options.map { it.productOptionId }.toList()
        }
    }

    data class Option(
        val productId: Long,
        val productOptionId: Long,
        val quantity: Int
    )
}
